platform/admin/core/admin-core-insecuredefaults/src/main/java/org/codice/ddf/admin/insecure/defaults/service/DefaultUsersDeletionScheduler.java

```diff
@@ -0,0 +1,234 @@
+      return true;
+    }
+  }
+
+  public static boolean removeDefaultUsers() {
+    try {
+      Bundle bundle = FrameworkUtil.getBundle(DefaultUsersDeletionScheduler.class);
+      if (bundle != null) {
+        BundleContext bundleContext = bundle.getBundleContext();
+        Collection<ServiceReference<BackingEngineFactory>> implementers =
+            bundleContext.getServiceReferences(BackingEngineFactory.class, null);
+        for (ServiceReference<BackingEngineFactory> impl : implementers) {
+
+          BackingEngineFactory backingEngineFactory = bundleContext.getService(impl);
+          BackingEngine backingEngine =
+              backingEngineFactory.build(
+                  ImmutableMap.of("users", getUsersPropertiesFilePath().toString()));
+
+          if (!backingEngine.listUsers().isEmpty()) {
+            backingEngine.listUsers().forEach(user -> backingEngine.deleteUser(user.getName()));
+          }
+        }
+      }
+
+      LOGGER.debug("Default users have been deleted successfully.");
+
+      Files.deleteIfExists(getTempTimestampFilePath());
+      return true;
+    } catch (Exception e) {
+      LOGGER.debug("Unable to remove default users.", e);
+      return false;
+    }
+  }
+
+  private String cronCalculator(Instant firstInstall) {
+    Instant threeDayTimestamp = firstInstall.plus(Duration.ofDays(3).minus(Duration.ofMinutes(30)));
+    LocalDateTime localDateTime =
+        LocalDateTime.ofInstant(threeDayTimestamp, ZoneId.systemDefault());
+
+    return String.format(
+        "%d+%d+%d+%d+%d+?+%d",
+        localDateTime.getSecond(),
+        localDateTime.getMinute(),
+        localDateTime.getHour(),
+        localDateTime.getDayOfMonth(),
+        localDateTime.getMonthValue(),
+        localDateTime.getYear());
+  }
+
+  public Instant installationDate() {
+    if (!getTempTimestampFilePath().toFile().exists()) {
+      return null;
+    }
+    try (ObjectInputStream objectInputStream =
+        new ObjectInputStream(new FileInputStream(getTempTimestampFilePath().toFile()))) {
+
+      return (Instant) objectInputStream.readObject();
+    } catch (IOException | ClassNotFoundException e) {
+      LOGGER.debug("Unable to read installation date.", e);
+      return null;
+    }
+  }
+
+  public void deleteScheduledDeletions() {
+    try {
+      if (context.getRoute(ROUTE_ID) != null
+          && context.getRouteStatus(ROUTE_ID).isStarted()
+          && context.getRouteStatus(ROUTE_ID).isStoppable()) {
+        context.stopRoute(ROUTE_ID);
+        Files.deleteIfExists(getTempTimestampFilePath());
+        LOGGER.debug("The deletion of default users has been stopped successfully.");
+      }
+    } catch (Exception e) {
+      LOGGER.debug("Unable to stop deletion.", e);
+    }
+  }
+
+  public boolean defaultUsersExist() {
+    try (Stream<String> defaultUsers = Files.lines(getUsersPropertiesFilePath())) {
+      return defaultUsers.filter(line -> !line.trim().isEmpty() && !line.startsWith("#")).count()
```

Comment:
✏️  For posterity, this could also be
```java
defaultUsers
  .filter(StringUtils::isNotBlank)
  .filter( line -> !line.startsWith("#")
  .anyMatch(line -> true)
```